frontend/app/(dashboard)/invoices/QuantityInput.tsx

```diff
@@ -13,31 +13,39 @@ const QuantityInput = ({
   onChange: (value: Value) => void;
 } & Omit<React.ComponentProps<"input">, "value" | "onChange">) => {
   const [rawValue, setRawValue] = useState("");
-  useEffect(
-    () => setRawValue(value ? (value.hourly ? formatDuration(value.quantity) : value.quantity.toString()) : ""),
-    [value],
-  );
 
-  return (
-    <Input
-      {...props}
-      value={rawValue}
-      onChange={(e) => setRawValue(e.target.value)}
-      onBlur={() => {
-        if (!rawValue.length) return onChange(null);
+  const parseRawValue = (raw: string): Value => {
+    if (!raw.length) return null;
+    const valueSplit = raw.split(":");
+    if (valueSplit.length === 1) return { quantity: parseFloat(valueSplit[0] ?? "0"), hourly: false };
+    const hours = parseInt(valueSplit[0] ?? "0", 10);
+    const minutes = parseInt(valueSplit[1] ?? "0", 10);
+    return {
+      quantity: Math.floor(isNaN(hours) ? 0 : hours * 60) + (isNaN(minutes) ? 0 : minutes),
+      hourly: true,
+    };
+  };
 
-        const valueSplit = rawValue.split(":");
-        if (valueSplit.length === 1) return onChange({ quantity: parseFloat(valueSplit[0] ?? "0"), hourly: false });
+  const onChangeLocal = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newRawValue = e.target.value;
+    setRawValue(newRawValue);
+    const parsedValue = parseRawValue(newRawValue);
+    onChange(parsedValue);
+  };
 
-        const hours = parseFloat(valueSplit[0] ?? "0");
-        const minutes = parseFloat(valueSplit[1] ?? "0");
-        onChange({
-          quantity: Math.floor(isNaN(hours) ? 0 : hours * 60) + (isNaN(minutes) ? 0 : minutes),
-          hourly: true,
-        });
-      }}
-    />
-  );
+  useEffect(() => {
+    const currentParsedValue = parseRawValue(rawValue);
+    const valuesEqual =
+      (currentParsedValue === null && value === null) ||
+      (currentParsedValue !== null &&
+        value !== null &&
+        currentParsedValue.quantity === value.quantity &&
+        currentParsedValue.hourly === value.hourly);
```

Comment:
```suggestion
      parsedValue?.quantity === value?.quantity &&
      parsedValue?.hourly === value?.hourly;
```
Shorter/simpler and I don't think we need the `current`